appverifier/src/main/java/com/android/mdl/appreader/fragment/ShowDocumentFragment.kt

```diff
@@ -236,6 +247,18 @@ class ShowDocumentFragment : Fragment() {
                     )
                 }
                 sb.append("</p><br>")
+                sb.append("<h6>Missing:</h6>")
+                sb.append("<p>")
+                // TODO: compare the original request with the received items, and show warning if necessary
+                if(portraitBytes==null){
+                    sb.append(
+                        "${
+                            getFormattedCheck(false)
+                        }<b>portrait</b> -> NULL<br>"
+                    )
+                    sb.insert(complianceAuthOffset, "${getFormattedCheck(false)}Provided portrait: NULL<br>")
+                }
+                sb.append("</p>")
```

Comment:
I think it's a little awkward to save `complianceAuthOffset` and then insert text at the bottom. Would be nicer to just have a helper method `isPortraitImagePresent()` and insert a warning at the top if - and only if - that helper returns false. Yes, this is another loop over the data but this is not a lot of data so it's fine to optimize for readability/clarity instead of CPU cycles spent.

As for the warning, I'm not sure "Provided portrait: NULL" is very meaningful. I think we should just in bold something human readable such as: _WARNING: No portrait image provided. This means it's not possible to verify the presenter is the authorized holder of the mDL._